### PR TITLE
[SDESK-4128] FunctionPoint service for generic function extensions

### PR DESCRIPTION
**Pull request by MarkLark86**
* Provide hook in AuthoringDirective.publishItem to process function extensions after validating and before publishing

### CODE REVIEW

**Comment by petrjasek** [comment]
hi, we were thinking about introducing similar api, but a bit different.
you can find a draft in https://gist.github.com/tomaskikutis/59646455c85a9ce61cea0c4e5665e843
and initial implementation in https://github.com/petrjasek/superdesk-client-core/commit/00e1a6adb594e23054fe1d02cadc72003c1f8e8d
we should decide on one approach and use it everywhere.
cc @tomaskikutis

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.ts [hunk 1]
```diff
@@ -425,7 +429,8 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
 
                 $scope.error = {};
 
-                return checkMediaAssociatedToUpdate()
+                return functionPoints.run('authoring:publish', {orig, item})
```
Comment (tomaskikutis) [inline scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: Could you only pass `item`, but not `orig`? Why do you need `orig`?

Comment (MarkLark86) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: `orig` and `item` are passed into the `publishItem` function, so I made the decision that the `FunctionPoint` should provide access to the same arguments from the parent function.

Comment (tomaskikutis) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.ts]: Got it. Please don't pass `orig` to `FunctionPoint` then. You can fetch the original from the server if you ever need it.

**Comment by MarkLark86** [comment]
@petrjasek Could we tag this one for v1.28 as well please? We need this for Planning and have v1.28 in production.

**Pull request merged**